fem/nonlinearform.hpp

```diff
@@ -35,6 +35,10 @@ class NonlinearForm : public Operator
    /// FE space on which the form lives.
    FiniteElementSpace *fes; // not owned
 
+   /** @brief Indicates the NonlinearForm%s stored in #dnfi,
+       #fnfi, #bfnfi and #bfnfi are owned by another NonlinearForm. */
+   int extern_bfs;
```

Comment:
```suggestion
   bool extern_bfs;
```
I know it's `int` in BilinearForm, but it really should be bool :)